app/controllers/follows_controller.rb

```diff
@@ -108,10 +110,19 @@ def follow(followable, need_notification: false)
     I18n.t("follows_controller.already_followed")
   end
 
+
+  def clear_followed_tag_caches
+    # Clear the followed_tags model cache, which is nested inside the async_info cache
+    Rails.cache.delete("#{current_user.cache_key}-#{current_user.last_followed_at&.rfc3339}/followed_tags")
```

Comment:
This is the _core_ problem — this cache [lives for most of a day](https://github.com/forem/forem/blob/main/app/decorators/user_decorator.rb#L47), so if it isn't cleared, it will appear for all intents & purposes that the following has not changed.